graphdatascience/query_runner/arrow_query_runner.py

```diff
@@ -79,9 +83,61 @@ def _run_arrow_property_get(self, graph_name: str, procedure_name: str, configur
         }
         ticket = flight.Ticket(json.dumps(payload).encode("utf-8"))
 
-        result: DataFrame = self._flight_client.do_get(ticket, self._flight_options).read_pandas()
+        get = self._flight_client.do_get(ticket)
+        result: DataFrame = get.read_pandas()
 
         return result
 
     def create_graph_constructor(self, graph_name: str, concurrency: int) -> GraphConstructor:
-        return ArrowGraphConstructor(self, graph_name, self._flight_client, self._flight_options, concurrency)
+        return ArrowGraphConstructor(self, graph_name, self._flight_client, concurrency)
+
+
+class AuthFactory(ClientMiddlewareFactory):  # type: ignore
+    def __init__(self, auth: Tuple[str, str], *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._auth = auth
+        self._token: Optional[str] = None
+        self._token_timestamp = 0
+
+    def start_call(self, info: Any) -> "AuthMiddleware":
+        return AuthMiddleware(self)
+
+    def token(self) -> Optional[str]:
+        # check whether the token is older than 10 minutes. If so, reset it.
+        if self._token and int(time.time()) - self._token_timestamp > 600:
+            self._token = None
+
+        return self._token
+
+    def set_token(self, token: str) -> None:
+        self._token = token
+        self._token_timestamp = int(time.time())
+
+    @property
+    def auth(self) -> Tuple[str, str]:
+        return self._auth
+
+
+class AuthMiddleware(ClientMiddleware):  # type: ignore
+    def __init__(self, factory: AuthFactory, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._factory = factory
+
+    def received_headers(self, headers: Dict[str, Any]) -> None:
+        auth_header: str = headers.get("Authorization", None)
+        if auth_header is not None:
+            [auth_type, token] = auth_header.split(" ", 1)
+            if auth_type == "Bearer":
+                self._factory.set_token(token)
```

Comment:
```suggestion
        if not auth_header:
            return
        [auth_type, token] = auth_header.split(" ", 1)
        if auth_type == "Bearer":
            self._factory.set_token(token)
```
(nitpick) prefer this style to avoid a lot of indentation